### PR TITLE
Create and Analyse Structural and Functional Networks of the Accounts Domain  

### PR DESCRIPTION
**Pull request by st-alice**
# Summary

This PR relates to creating and exploring the structural and functional networks of pages that traverse to/from `www.gov.uk` from/to `signin.account.gov.uk` and `account.gov.uk`. Data is prepped and analysed following network analysis/graph theory processes. NOTE: to create the functional network you must be able to use Google Cloud Platform through code on your local machine and have the correct permissions to access `govuk-bigquery-analytics` project. See: https://docs.data-community.publishing.service.gov.uk/analysis/google-cloud-platform/#use-gcp-through-the-command-line-on-your-local-machine

There are 3 scripts for review: 2 function scripts and a notebook. 

- `notebooks/create_networks.ipynb`: A notebook which can be run to create, *1) *the structural network, *2)* the functional network, *3)* simple comparisons between the functional and structural networks, *4)* a bokeh plot visualising the differences between the structural and functional network 
 
- `src/make_data/create_networks.py`: Functions which create the functional network. 

- `src/make_visualisations/create_network_plot.py`: Function for the interactive Bokeh plot to visualise observed user movements across the structural network. 

Creating and comparing the networks is the first step for extracting the intended journeys. 

# Checklists

This pull/merge request meets the following requirements:

- [X] code runs
- [X] [developments are ethical][data-ethics-framework] and secure
- [X] you have made proportionate checks that the code works correctly
- [X] developments adhere to AQA plan (see `docs/aqa/aqa_plan.md`)
- [ ] ~~assumptions, and caveats log updated (see `docs/aqa/assumptions_caveats.md`), if
  necessary~~ I will do this later on once the POC is approved. Currently, I am keeping track of assumptions on each script.  
- [ ] ~~[minimum usable documentation][agilemodeling] written in the `docs` folder~~ Also not needed until POC is approved.

Comments have been added below around the incomplete checks.

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

File: src/make_data/create_networks.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        df.sort_values(by=["sessionId", "hitNumber"], ascending=True)
+        .groupby(["sessionId"])["sourcePagePath"]
+        .shift(-1)
+    )
+
+    df = df[["sessionId", "sourcePagePath", "destinationPagePath"]]
+
+    # Count the number of user sessions that traverse from a sourcePagePath to
+    # a destinationPagePath
+    df = (
```
Comment (zilnhoj) [inline src/make_data/create_networks.py:200]: maybe use a new dataframe variable - there may be issues with over writing variables

File: src/make_visualisations/create_network_plot.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
+import networkx as nx
+from bokeh.io import show
+from bokeh.models import Circle, MultiLine, NodesAndLinkedEdges
+from bokeh.plotting import figure, from_networkx
+
+
+def create_bokeh_plot(
```
Comment (zilnhoj) [inline src/make_visualisations/create_network_plot.py:7]: Love that you are using Bokeh - not for now but i've come across a way to host a bokeh server on GCP and we may be able to use that as an output for this down the line

File: src/make_data/create_networks.py (hunk 1 above)
Comment (st-alice) [reply src/make_data/create_networks.py]: Thanks @zilnhoj - good point. I have updated to a new variable name in commit 823ce16aeb313aa2e646a57c0f4b6fb6f462fda0

File: src/make_visualisations/create_network_plot.py (hunk 2 above)
Comment (st-alice) [reply src/make_visualisations/create_network_plot.py:7]: Thanks @zilnhoj , I will keep this comment open for future reference!

**Pull request merged**